python/lsst/ip/isr/isrMockLSST.py

```diff
@@ -0,0 +1,597 @@
+        dtype=bool,
+        default=True,
+        doc="If True, products have one raw image per amplifier, otherwise, one raw image per detector.",
+    )
+    # Signal parameters.
+    # Most of them are inherited from isrMockConfig but we update
+    # some to LSSTcam expected values.
+    # TODO: DM-42880 Update values to what is expected in LSSTCam.
+    biasLevel = pexConfig.Field(
+        dtype=float,
+        default=30000.0,
+        doc="Background contribution to be generated from the bias offset in ADU.",
+    )
+    # Inclusion parameters are inherited from isrMock.
+    doAddParallelOverscan = pexConfig.Field(
+        dtype=bool,
+        default=True,
+        doc="Add overscan ramp to parallel overscan and data regions.",
+    )
+    doAddSerialOverscan = pexConfig.Field(
+        dtype=bool,
+        default=True,
+        doc="Add overscan ramp to serial overscan and data regions.",
+    )
+    doAddGain = pexConfig.Field(
+        dtype=bool,
+        default=True,
+        doc="Add gain to data.",
+    )
+
+
+class IsrMockLSST(IsrMock):
+    """Class to generate consistent mock images for ISR testing.
+
+    ISR testing currently relies on one-off fake images that do not
+    accurately mimic the full set of detector effects. This class
+    uses the test camera/detector/amplifier structure defined in
+    `lsst.afw.cameraGeom.testUtils` to avoid making the test data
+    dependent on any of the actual obs package formats.
+    """
+    ConfigClass = IsrMockLSSTConfig
+    _DefaultName = "isrMockLSST"
+
+    def __init__(self, **kwargs):
+        # cross-talk coeffs, bf kernel are defined in the parent class.
+        super().__init__(**kwargs)
+
+    def run(self):
+        """Generate a mock ISR product following LSSTCam ISR, and return it.
+
+        Returns
+        -------
+        image : `lsst.afw.image.Exposure`
+            Simulated ISR image with signals added.
+        dataProduct :
+            Simulated ISR data products.
+        None :
+            Returned if no valid configuration was found.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if both doGenerateImage and doGenerateData are specified.
+        """
+        if self.config.doGenerateImage and self.config.doGenerateData:
+            raise RuntimeError("Only one of doGenerateImage and doGenerateData may be specified.")
+        elif self.config.doGenerateImage:
+            return self.makeImage()
+        elif self.config.doGenerateData:
+            return self.makeData()
+        else:
+            return None
+
+    def makeImage(self):
+        """Generate a simulated ISR LSST image.
+
+        Returns
+        -------
+        exposure : `lsst.afw.image.Exposure` or `dict`
+            Simulated ISR image data.
+
+        Notes
+        -----
+        This method constructs a "raw" data image.
+        """
+        exposure = self.getExposure()
+
+        for idx, amp in enumerate(exposure.getDetector()):
+
```

Comment:
I would suggest adding some comments here about what we're doing... adding in effects "forward" as they come into the telescope, and then ISR will reverse these.  That's why we start in electrons and end in ADU.